From f7236c817aee8c39223cca88deb8341b4f2b5dfa Mon Sep 17 00:00:00 2001
From: Cathy Avery <cavery@redhat.com>
Date: Mon, 18 Dec 2023 14:13:19 -0500
Subject: settings.py: update settings for rhel

    commit 2bf34313f2e9599e3304b5446411b5ada6ccd7f0
    Author: Ani Sinha <anisinha@redhat.com>
    Date:   Tue Apr 11 04:20:00 2023 -0400
    Please see commit 5e1e568d7085fd4443

(" Add initial redhat setup")
from rhel8.8.0 branch for setings.py. Applying the same for the rebased
cloud-init.

X-downstream-only: true

Signed-off-by: Ani Sinha <anisinha@redhat.com>

Conflicts: "def_log_file_mode": 0o600 as commit
130899115 'limit permissions on def_log_file' was not applied.

Signed-off-by: Cathy Avery <cavery@redhat.com>
---
 cloudinit/settings.py            |  7 +++++--
 tests/unittests/cmd/test_main.py | 17 +++++++++++------
 2 files changed, 16 insertions(+), 8 deletions(-)

diff --git a/cloudinit/settings.py b/cloudinit/settings.py
index 592e144d..87829ff0 100644
--- a/cloudinit/settings.py
+++ b/cloudinit/settings.py
@@ -54,13 +54,16 @@ CFG_BUILTIN = {
     ],
     "def_log_file": "/var/log/cloud-init.log",
     "log_cfgs": [],
-    "syslog_fix_perms": ["syslog:adm", "root:adm", "root:wheel", "root:root"],
+    "syslog_fix_perms": [],
+    "mount_default_fields": [None, None, "auto", "defaults,nofail", "0", "2"],
+    "ssh_deletekeys": False,
+    "ssh_genkeytypes": [],
     "system_info": {
         "paths": {
             "cloud_dir": "/var/lib/cloud",
             "templates_dir": "/etc/cloud/templates/",
         },
-        "distro": "ubuntu",
+        "distro": "rhel",
         "network": {"renderers": None},
     },
     "vendor_data": {"enabled": True, "prefix": []},
diff --git a/tests/unittests/cmd/test_main.py b/tests/unittests/cmd/test_main.py
index ab427115..c8c2ae81 100644
--- a/tests/unittests/cmd/test_main.py
+++ b/tests/unittests/cmd/test_main.py
@@ -119,14 +119,19 @@ class TestMain(FilesystemMockingTestCase):
                 {
                     "def_log_file": "/var/log/cloud-init.log",
                     "log_cfgs": [],
-                    "syslog_fix_perms": [
-                        "syslog:adm",
-                        "root:adm",
-                        "root:wheel",
-                        "root:root",
-                    ],
                     "vendor_data": {"enabled": True, "prefix": []},
                     "vendor_data2": {"enabled": True, "prefix": []},
+                    "syslog_fix_perms": [],
+                    "ssh_deletekeys": False,
+                    "ssh_genkeytypes": [],
+                    "mount_default_fields": [
+                        None,
+                        None,
+                        "auto",
+                        "defaults,nofail",
+                        "0",
+                        "2",
+                    ],
                 }
             )
             updated_cfg.pop("system_info")
-- 
2.41.0

